docs/adr/ADR-7_wsgi-server.md

```diff
@@ -0,0 +1,298 @@
+# ADR 7: WSGI Server
+
+## Status
+
+Proposed
+
+## Context
+
+We were running the Flask WSGI application with the built-in Flask server, which is not meant for production. Problems are mainly due to potential debug shell on the server and single thread in default configuration. Both were mitigated in our setup, but we decided to test a proper WSGI server at some point. Especially the log entry
+
+```text
+ * Serving Flask app 'connaisseur.flask_server' (lazy loading)
+ * Environment: production
+   WARNING: This is a development server. Do not use it in a production deployment.
+   Use a production WSGI server instead.
+```
+did cause anguish among users, see e.g. [issue 11](https://github.com/sse-secure-systems/connaisseur/issues/11).
+
+## Considered options
+
+### Choice 1: WSGI server
+
+There's plenty of WSGI server around and the question poses itself, which one to pick. Flask itself has a [list of servers](https://flask.palletsprojects.com/en/1.1.x/deploying/), there's comparisons around, for example [here](https://medium.com/django-deployment/which-wsgi-server-should-i-use-a70548da6a83) and [here](https://www.appdynamics.com/blog/engineering/a-performance-analysis-of-python-wsgi-servers-part-2/). The choice, which WSGI servers to test was somewhat arbitrary among better performing ones in the posts.
+
+Contenders were Bjoern, Cheroot, Flask, Gunicorn and uWSGI. Bjoern was immediately dropped, since it works only with Python2. Gunicorn was tested for a bit, but since it delivered worse results than the others and it requires a writable log directory, it was also dropped from contention.
+
+The remaining three were tested over a rather long of development, i.e. from before the first bit of validation parallelization to after the 2.0 release. All tests were run on local minikube/kind clusters with rather constrained resources in the expectation that this will still provide reasonable insight into the servers' behaviour on regular production clusters.
```

Comment:
```suggestion
The remaining three were tested over a rather long time of development, i.e. from before the first bit of validation parallelization to after the 2.0 release. All tests were run on local minikube/kind clusters with rather constrained resources in the expectation that this will still provide reasonable insight into the servers' behavior on regular production clusters.
```